// dataservice/lib/dataservice.js
/**
 * Created with JetBrains WebStorm.
 * User: Chris
 * Date: 3/12/13
 * Time: 6:06 AM
 * To change this template use File | Settings | File Templates.
 */
var events = require("events")
    ,util = require("util");

var DataService = function(options){
    $this = this;
    events.EventEmitter.call($this);
    $this.api = options.api;
    $this.mapper = options.mapper;
}
util.inherits(DataService,events.EventEmitter);
DataService.prototype.events = {
    getUsers:'getUsers',
    getNewUser:'getNewUser',
    saveUser:'saveUser'
}
DataService.prototype.getUsers = function(options){
    options = options || {};
     var callback = function(result){
        var result = $this.mapper.map('users',result);
        $this.emit($this.events.getUsers,result);
     }
    options.callback = callback;
    var data = this.api.getUsers(options);

}

DataService.prototype.getNewUser=function(options){
    options = options || {};
     var callback = function(result){
        var result = $this.mapper.map('user',result);
        $this.emit($this.events.getNewUser,result);
     }
    options.callback = callback;
    var data = this.api.getNewUser(options);
}
DataService.prototype.saveUser=function(options){
    options = options || {};
     var callback = function(result){
        var result = $this.mapper.map('user',result);
        $this.emit($this.events.saveUser,result);
     }
    options.callback = callback;
    var data = this.api.saveUser(options);
}

module.exports = DataService;
